main.go

```diff
@@ -35,6 +35,38 @@ type Entropy struct {
 	Line    string  // Line with high entropy
 }
 
+// Entropies should be created with a size n using make()
+// it should not be written to manually, instead use Entropies.Add
+type Entropies struct {
+	sync.Mutex
+	Entropies []Entropy
+}
+
+// Add assumes that es contains an ordered set of entropies.
+// It preserves ordering, and inserts an additional value e, if it has high enough entropy.
+// In that case, the entry with lowest entropy is rejected.
+func (es *Entropies) Add(e Entropy) {
+	es.Lock()
+	defer es.Unlock()
+
+	if es.Entropies[len(es.Entropies)-1].Entropy >= e.Entropy {
+		return
+	}
+
+	i, _ := slices.BinarySearchFunc(es.Entropies, e, func(a, b Entropy) int {
+		if b.Entropy > a.Entropy {
+			return 1
+		}
+		if a.Entropy > b.Entropy {
+			return -1
+		}
+		return 0
+	})
+
+	copy(es.Entropies[i+1:], es.Entropies[i:])
+	es.Entropies[i] = e
```

Comment:
I think we need `if i < len(es.Entropies)` check here as added item may evict the last one (i == len(es.Entropies))

Reply:
We don´t need this, when we are evicting the last element, the `copy` simply becomes a no-op.

The last item is `i == len(es.Entropies)-1`, not `i == len(es.Entropies)` as well.